compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1356,6 +1356,14 @@ impl<'tcx> TyCtxt<'tcx> {
         self.diagnostic_items(did.krate).name_to_id.get(&name) == Some(&did)
     }
 
+    /// Returns `Some(async_generator_kind)` if the node pointed to by `def_id` is an async generator.
+    pub fn async_kind(self, def_id: DefId) -> Option<hir::AsyncGeneratorKind> {
```

Comment:
This seems way too similar to the `TyCtxt` query `fn asyncness(self, def_id: DefId) -> hir::IsAsync`. Can you change the name to be a bit more generator-specific? Perhaps `generator_is_async` or something.

Also, @fbstj does have a good point that should probably just return a boolean (or a newtyped boolean).

Reply:
If it's gonna keep the signature with the `Option`, then perhaps `generator_asyncness` or `generator_async_kind` is a better name.